src/main/java/com/microsoft/azure/relay/HybridConnectionListener.java

```diff
@@ -696,6 +714,10 @@ else if (command.getRenewToken() != null) {
 
 				ClientWebSocket webSocket = new ClientWebSocket(this.listener.trackingContext, EXECUTOR);
 				return delayTask.thenCompose(($void) -> {
+	                if (this.listener.injectedFault != null && this.listener.injectedFault instanceof UpgradeException) {
```

Comment:
>this.listener.injectedFault instanceof UpgradeExceptio [](start = 60, length = 54)

injectedFault seems to only ever get set if the exception is-an UgradeException. With the current code I can't see any way for this instanceof check to ever return false.

Reply:
Sounds like you want to keep it generic for future use. ok. Close/resolve/whatever

---
In reply to: [354521815](https://github.com/Azure/azure-relay-java/pull/58#discussion_r354521815) [](ancestors = 354521815)